pkg/daemon/ceph/osd/kms/vault.go

```diff
@@ -124,13 +127,50 @@ func configTLS(clusterdContext *clusterd.Context, namespace string, config map[s
 
```

Comment:
If the func fails to write a file above, where does it clean up by removing the files when it returns an err?

Would it be better to always try to remove the files before starting the write process?

Reply:
>If the func fails to write a file above, where does it clean up by removing the files when it returns an err?

Hum, if the write fails, nothing is written right?

>Would it be better to always try to remove the files before starting the write process?

We always get a new tmp file, so we have nothing to forcibly remove before running.

Reply:
But this is done in a loop right and there can be multiple files created like this? Should this function remove successfully created files if it returns with a failure?